services/rsources/extension.go

```diff
@@ -0,0 +1,130 @@
+package rsources
+
+import (
+	"context"
+	"database/sql"
+	"errors"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/lib/pq"
+	"github.com/rudderlabs/rudder-server/config"
+)
+
+type extension interface {
+	getReadDB() *sql.DB
+	createStatsTable(ctx context.Context, jobRunId string) error
+	dropTables(ctx context.Context, jobRunId string) error
+	cleanupLoop(ctx context.Context) error
+}
+
+type defaultExtension struct {
+	localDB *sql.DB
+}
+
+func (r *defaultExtension) getReadDB() *sql.DB {
+	return r.localDB
+}
+
+func (r *defaultExtension) createStatsTable(ctx context.Context, jobRunId string) error {
+	dbHost := config.GetEnv("JOBS_DB_HOST", "localhost")
+	jobRunIdStatTableName := fmt.Sprintf("%s%s", tablePrefix, jobRunId)
+	sqlStatement := fmt.Sprintf(`create table if not exists "%s" (
+		db_name text not null default '%s',
+		source_id text not null,
+		destination_id text not null,
+		task_run_id text not null,
+		in_count integer not null default 0,
+		out_count integer not null default 0,
+		failed_count integer not null default 0,
+		ts timestamp not null default NOW(),
+		primary key (db_name, source_id, destination_id, task_run_id)
+	)`, jobRunIdStatTableName, dbHost)
+	_, err := r.localDB.ExecContext(ctx, sqlStatement)
+	return err
+}
+
+func (r *defaultExtension) cleanupLoop(ctx context.Context) error {
+	err := r.doCleanupTables(ctx)
+	if err != nil {
+		return err
+	}
+	for {
+		select {
+		case <-ctx.Done():
+			return nil
+		case <-time.After(sourcesStatCleanUpSleepTime):
+			err := r.doCleanupTables(ctx)
+			if err != nil {
+				return err
+			}
+		}
+	}
+}
+
+func (r *defaultExtension) doCleanupTables(ctx context.Context) error {
+	db := r.localDB
+	tables, err := getAllTableNames(ctx, db)
+	if err != nil {
+		return err
+	}
+
+	before := time.Now().Add(-24 * time.Hour)
```

Comment:
I believe this should be configurable

Reply:
simply replaced it with `config.GetDuration("JOBS_CLEANUP_INTERVAL", 24, time.Hour)`